src/WidgetBase.ts

```diff
@@ -291,6 +293,33 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 		return this._properties;
 	}
 
+	@diffProperty('registry', reference)
+	protected diffPropertyRegistry(previousProperties: any, newProperties: any): void {
+		if (this._registries.defaultRegistry === this._defaultRegistry) {
+			this._registries.remove(this._defaultRegistry);
+		}
+		const result = this._registries.replace(previousProperties.registry, newProperties.registry);
```

Comment:
Currently, out of the box uglify would leave a `this.whatever` alone, since it cannot be renamed, since it does not have a local variable to deal with.  Changing it to a local variable, when you have multiple access to it, will allow uglify to shorten the variable, so something like this:

```js
const whatever = this.whatever;
if (!this.whatever.has()) {
    this.whatever.add({});
}
```

can be re-written to:

```js
const a=this.whatever if(!a.has()){a.add({})}
```

Instead of:

```
if(!this.whatever.has()){this.whatever.add({})}
```

It really only matters when you have several access to the same member in a block.  There is also a small small performance gain of lookup, though most runtimes would optimise that anyways.